### PR TITLE
Modified effective area format. EFFAREA_RECO column moved to a new HDU.

### PR DESCRIPTION
**Pull request by TarekHC**
As discussed in issue #35 , I added some changes to the effective area format. 

Essentially, I removed the  ``EFFAREA_RECO`` column, and described the effective area vs estimated energy as a separate HDU.

The format may be definitely improved. I'm not sure if the "``aeff_2d`` format" should refer to both effective areas, or they are "different formats".

I also modified the python macro that generates the FITS file example, updating it.

### CODE REVIEW

**Comment by cdeil** [comment]
@jknodlseder - Do you have time to review this?

From my side, I think the main comment would be that there should be a declarative way to tell which AEFF a given HDU is. I guess this means that the `HDUCLAS2` header keyword should be different for the two types of AEFF?

File: source/irfs/effective_area/index.rst [hunk 1]
```diff
@@ -54,4 +52,34 @@ sources.
...
     * On region radius for point-like observations
 
+Although not a requirement, the recommended ``EXTNAME`` keyword is "EFFECTIVE AREA".
+
+.. _aeff_reco_2d:
+
+Effective Area vs reconstructed energy
+--------------------------------------
+
+Although not required, it is generally useful to store the effective area as a function of the reconstructed energy,
```
Comment (cdeil) [inline source/irfs/effective_area/index.rst]: > Although not required, it is generally useful

Can you rephrase here?

Both (AEFF, EDISP) and the combined AEFF_RECO are in use for current IACTs and both are useful.
Whether CTA could just pick one and never need the other is unclear to me ... my guess would be no, i.e. both will be used.

* Using AEFF, EDISP is more precise, but it requires enough MC stats to have an EDISP with acceptable noise, and it's slower to use in likelihood fits for science tools.
* Using AEFF_RECO is less precise, requires less MC stats and is faster to use in likelihood fits for science tools.

Maybe it's OK to extend the description of this format to explain this briefly why they both exist and when to use which one as DL3 producer (i.e. roughly the short summary I wrote here)?

**Comment by TarekHC** [comment]
Thanks @cdeil ,

>From my side, I think the main comment would be that there should be a declarative way to tell which AEFF a given HDU is. I guess this means that the HDUCLAS2 header keyword should be different for the two types of AEFF?

As the HDUCLAS2 was not present there before, I decided to not include it. I will simply add it to both effective areas description.

> Maybe it's OK to extend the description of this format to explain this briefly why they both exist and when to use which one as DL3 producer (i.e. roughly the short summary I wrote here)?

Ok, verbose mode on. I will commit those changes today.

**Comment by TarekHC** [comment]
Ok, done. I tried to implement all comments. I also re-structured the text a bit. Let me know if you think I should change anything else.

**Comment by cdeil** [comment]
@TarekHC - Thanks!

Assigning @jknodlseder for review / decision.
I'm going on holiday for two weeks and am out of time. 
Feel free to merge, don't wait for comments from me.

**Comment by TarekHC** [comment]
@cdeil @jknodlseder , we are all extremely busy with Kashiwa... So there is really no rush. Just review it whenever you find the time.

**Comment by cdeil** [comment]
@TarekHC - Can you please rebase this pull request against master? (Github shows that it can't be merged because there is a conflict).

@jknodlseder - Do you have time to review this PR this week?

From my side it's OK ... the main point is that it splits `EFF_AREA_RECO` in a separate HDU now and there are `HDUCLAS2` header keywords to declare what kind of response it is.

**Comment by TarekHC** [comment]
Yes, no problem. I'll take care of that. Although I will probably wait for jurgen's feedback first, as new conflicts may arise.

**Comment by cdeil** [comment]
@TarekHC - Can you please rebase this PR now?

Background info: I'd like to merge this (and the other open PRs) now, so that further work on the spec isn't blocked. We'd like to clean up several details for the next HESS FITS production that will then be used for the test data release.

**Comment by TarekHC** [comment]
Oh, sure. I'll try to do it today then. Cheers!

**Comment by TarekHC** [comment]
Ok, I think the conflicts were solved.
